### PR TITLE
Pass translation key for login and auth prompts

### PR DESCRIPTION
**Pull request by llewelld**
Passes the gecko translation keys along with the form data to the browser UI components for auth and login dialogs. This allows localization to be performed in the UI.

### CODE REVIEW

**Comment by llewelld** [comment]
Requires https://github.com/sailfishos/sailfish-components-webview/pull/117

**Comment by llewelld** [comment]
@okodron the settings still need to be tweaked so I can add you as a reviewer here, but your review would be appreciated.

File: jscomps/PromptService.js [hunk 1]
```diff
@@ -723,7 +772,20 @@ var PromptUtils = {
       realm += this.ellipsis;
     }
 
-    return realm;
+    let text;
```
Comment (chriadam) [inline jscomps/PromptService.js:775]: nitpick: I think I would prefer "textKey" as the variable name

Comment (llewelld) [reply jscomps/PromptService.js]: Yeah, agreed, it's better to make it clear.

**Comment by llewelld** [comment]
I've updated the PRs to send the localization key and parameter inputs as an array bundle, so they can be interpreted in a generic way on the QML side.

This should now be ready for review so I'm removing the WIP prefix.

**Comment by llewelld** [comment]
I've added the commits for "JB#54829/OMP#JOLLA-221 - Find outstanding gecko strings for conversion to Qt id" on top of this one since it relates.

File: jscomps/PromptService.js [hunk 2]
```diff
@@ -371,7 +423,8 @@ InternalPrompt.prototype = {
   },
 
   select: function select(aTitle, aText, aCount, aSelectList, aOutSelection) {
-    let p = this._getPrompt(aTitle, aText, [ PromptUtils.getLocaleString("OK") ]);
+    let p = this._getPrompt(aTitle, aText, [ "OK" ]);
+    p.setHint("select");
```
Comment (rainemak) [inline jscomps/PromptService.js:427]: Found it: https://github.com/sailfishos/sailfish-components-webview/pull/117#discussion_r666758477

I'd not introduce back blocking "embed:select" if not mandatory. With newer engine nsIPromptService.idl has asyncSelect as well.

Comment (rainemak) [reply jscomps/PromptService.js:427]: As you commented, this is only for hybrid app developers. Hence, can be added here for sake of completeness.

**Review by chriadam — APPROVED** [review]

**Review by okodron — APPROVED** [review]

**Comment by rainemak** [comment]
Remember to rebase before merging.

**Comment by rainemak** [comment]
Superseded by https://github.com/sailfishos/embedlite-components/pull/23

**Pull request closed without merging**